wagtail/snippets/bulk_actions/delete.py

```diff
@@ -1,19 +1,51 @@
+from django.utils.functional import cached_property
 from django.utils.text import capfirst
 from django.utils.translation import gettext_lazy as _
 from django.utils.translation import ngettext
 
+from wagtail.admin.views.generic import BeforeAfterHookMixin
 from wagtail.snippets.bulk_actions.snippet_bulk_action import SnippetBulkAction
 from wagtail.snippets.permissions import get_permission_name
 
 
-class DeleteBulkAction(SnippetBulkAction):
+class DeleteBulkAction(BeforeAfterHookMixin, SnippetBulkAction):
     display_name = _("Delete")
     action_type = "delete"
     aria_label = _("Delete selected snippets")
     template_name = "wagtailsnippets/bulk_actions/confirm_bulk_delete.html"
     action_priority = 30
     classes = {"serious"}
 
+    @cached_property
+    def _actionable_objects(self):
```

Comment:
This is a bit tangled but I recommend reading the base bulk action view to see why I wrote it this way. In short, the base view's override points aren't as good as they could be.